Tests Fail With Some Pandas / Avro Combinations

- OS type and version: Linux
  - Python version: 3.9
  - `google-cloud-bigquery-storage` version: 2.14.1
  - `fastavro` version: 0.21.4
  - `pandas` version: 1.1.5

Running the test suite with these versions of fastavro and pandas, which look like they're supported version according to the setup.py, result in the following test failure:

```
tests/unit/test_reader_v1.py ...........F....

=================================== FAILURES ===================================
_________________________ test_to_dataframe_w_scalars __________________________

class_under_test = <class 'google.cloud.bigquery_storage_v1.reader.ReadRowsStream'>
mock_gapic_client = <MagicMock spec='BigQueryReadClient' id='139834579968976'>

    def test_to_dataframe_w_scalars(class_under_test, mock_gapic_client):
        avro_schema = _bq_to_avro_schema(SCALAR_COLUMNS)
        avro_blocks = _bq_to_avro_blocks(SCALAR_BLOCKS, avro_schema)
    
        mock_gapic_client.read_rows.return_value = avro_blocks
        reader = class_under_test(mock_gapic_client, "", 0, {})
        got = reader.to_dataframe()
    
        expected = pandas.DataFrame(
            list(itertools.chain.from_iterable(SCALAR_BLOCKS)), columns=SCALAR_COLUMN_NAMES
        )
        # fastavro provides its own UTC definition, so
        # compare the timestamp columns separately.
        got_ts = got["ts_col"]
        got = got.drop(columns=["ts_col"])
        expected_ts = expected["ts_col"]
        expected = expected.drop(columns=["ts_col"])
    
        pandas.testing.assert_frame_equal(
            got.reset_index(drop=True),  # reset_index to ignore row labels
            expected.reset_index(drop=True),
        )
>       pandas.testing.assert_series_equal(
            got_ts.reset_index(drop=True),
            expected_ts.reset_index(drop=True),
            check_dtype=False,  # fastavro's UTC means different dtype
            check_datetimelike_compat=True,
        )
E       AssertionError: [datetimelike_compat=True] <DatetimeArray>
E       ['2000-01-01 05:00:00+00:00', '1965-04-03 02:01:00+00:00',
E        '1991-08-25 20:57:08+00:00']
E       Length: 3, dtype: datetime64[ns, <fastavro._timezone.UTCTzinfo object at 0x7f2ddd7a5040>] is not equal to <DatetimeArray>
E       ['2000-01-01 05:00:00+00:00', '1965-04-03 02:01:00+00:00',
E        '1991-08-25 20:57:08+00:00']
E       Length: 3, dtype: datetime64[ns, UTC].

tests/unit/test_reader_v1.py:478: AssertionError
```

This assertion fails:

```python
  pandas.testing.assert_series_equal(                                                                                                                                                                                                                           
      got_ts.reset_index(drop=True),                                                                                                                                                                                                                            
      expected_ts.reset_index(drop=True),                                                                                                                                                                                                                       
      check_dtype=False,  # fastavro's UTC means different dtype                                                                                                                                                                                                
      check_datetimelike_compat=True,                                                                                                                                                                                                                           
  ) 
```

The test succeeds with:

```python
check_datetimelike_compat=False
```

or by setting the dtype on the expected series:

```python
pandas.testing.assert_series_equal(
    got_ts.reset_index(drop=True),
    expected_ts.astype(got_ts.dtype).reset_index(drop=True)
)
```

--- a/tests/unit/test_reader_v1.py
+++ b/tests/unit/test_reader_v1.py
@@ -477,9 +477,7 @@ def test_to_dataframe_w_scalars(class_under_test, mock_gapic_client):
     )
     pandas.testing.assert_series_equal(
         got_ts.reset_index(drop=True),
-        expected_ts.reset_index(drop=True),
-        check_dtype=False,  # fastavro's UTC means different dtype
-        check_datetimelike_compat=True,
+        expected_ts.reset_index(drop=True).astype(got_ts.dtype),
     )
 
 

